### PR TITLE
feat: Notification page e2e

### PR DESCRIPTION
**Pull request by LinMoskovitch**

### CODE REVIEW

File: Notify/Notify/Notify/Helpers/Converter.cs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/Helpers/Converter.cs:3]: Why do we need to use it, and not our customized logger?

File: Notify/Notify/Notify/Helpers/Converter.cs [hunk 2]
```diff
@@ -19,6 +20,107 @@ public object ConvertBack(object value, Type targetType, object parameter, Cultu
...
+                    case NotificationType.Dynamic:
+                        return ImageSource.FromFile("dynamic_location_colored_icon");
+                    case NotificationType.Time:
+                        return ImageSource.FromFile("time_colored_icon");
+                    default:
+                        return null; 
+                }
+            }
+            
+            Debug.WriteLine($"Unexpected value in TypeToIconConverter: {value?.GetType()} - {value}");
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/Helpers/Converter.cs:114]: Why do we need to use it, and not our customized logger?

File: Notify/Notify/Notify/ViewModels/NotificationsPageViewModel.cs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 using System;
 using System.Collections.Generic;
 using System.ComponentModel;
+using System.Diagnostics;
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/ViewModels/NotificationsPageViewModel.cs:4]: Why do we need to use it, and not our customized logger?

File: Notify/Notify/Notify/ViewModels/NotificationsPageViewModel.cs [hunk 4]
```diff
@@ -45,63 +92,110 @@ public NotificationsPageViewModel()
...
+        {
+            if (EqualityComparer<T>.Default.Equals(field, value)) return;
+            field = value;
+            OnPropertyChanged(propertyName);
+        }
+        
+        private async void onDeleteNotificationButtonClicked(Notification notification)
+        {
+            bool isDeleted;
+            Debug.WriteLine("Delete notification button clicked");
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/ViewModels/NotificationsPageViewModel.cs:130]: Why do we need to use it, and not our customized logger?

File: Notify/Notify/Notify/ViewModels/NotificationsPageViewModel.cs [hunk 5]
```diff
@@ -45,63 +92,110 @@ public NotificationsPageViewModel()
...
+                        "Failed to delete notification",
+                        "OK");
+                }
+            }
+        }
 
-        private void OnPropertyChanged([CallerMemberName] string propertyName = null)
+        private async void onEditNotificationButtonClicked(Notification notification)
+        {
+            Debug.WriteLine("edit notification button clicked");
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/ViewModels/NotificationsPageViewModel.cs:154]: Why do we need to use it, and not our customized logger?

File: Notify/Notify/Notify/ViewModels/NotificationsPageViewModel.cs [hunk 6]
```diff
@@ -45,63 +92,110 @@ public NotificationsPageViewModel()
...
         }
 
-        private bool SetField<T>(ref T field, T value, [CallerMemberName] string propertyName = null)
+        private async void onRenewNotificationButtonClicked(Notification notification)
         {
-            if (EqualityComparer<T>.Default.Equals(field, value)) return false;
-            field = value;
-            OnPropertyChanged(propertyName);
-            return true;
+            Debug.WriteLine("renew notification button clicked");
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/ViewModels/NotificationsPageViewModel.cs:165]: Why do we need to use it, and not our customized logger?

File: Notify/Notify/Notify/Helpers/Converter.cs [hunk 7]
```diff
@@ -19,6 +20,107 @@ public object ConvertBack(object value, Type targetType, object parameter, Cultu
...
+                switch (type)
+                {
+                    case NotificationType.Location:
+                        return ImageSource.FromFile("location_colored_icon"); 
+                    case NotificationType.Dynamic:
+                        return ImageSource.FromFile("dynamic_location_colored_icon");
+                    case NotificationType.Time:
+                        return ImageSource.FromFile("time_colored_icon");
+                    default:
+                        return null;
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/Helpers/Converter.cs:110]: I think that an exception should be thrown here, for passing invalid argument.

**Review by OfirMatasas — CHANGES_REQUESTED** [review]
Overall looks great!
Just a few changes.

File: Notify/Notify/Notify/Helpers/Converter.cs (hunk 2 above)
Comment (LinMoskovitch) [reply Notify/Notify/Notify/Helpers/Converter.cs:114]: Actually I don't need this anymore, I had trouble with it at the beginning.

File: Notify/Notify/Notify/Helpers/Converter.cs (hunk 7 above)
Comment (LinMoskovitch) [reply Notify/Notify/Notify/Helpers/Converter.cs:110]: RIght

**Review by OfirMatasas — APPROVED** [review]
LGTM

**Pull request merged**